readthedocs/restapi/views/model_views.py

```diff
@@ -215,6 +215,17 @@ def sync_versions(self, request, **kwargs):  # noqa: D205
                 status=status.HTTP_400_BAD_REQUEST,
             )
 
+        try:
+            api_utils.run_automation_rules(project, added_versions)
+        except Exception:
+            # Don't interrupt the request if something goes wrong
+            # in the automation rules.
+            log.exception(
+                'Failed to execute automation rules for [%s]: %s',
+                project.slug, added_versions
+            )
+
+        # TODO: move this to an automation rule
```

Comment:
I wanted to move this to the `run_automation_rules` function, but it depends on the previous state of one version. It can be refactored for later